composeApp/src/commonMain/kotlin/org/ooni/probe/ui/settings/proxy/ProxyScreen.kt

```diff
@@ -0,0 +1,228 @@
+                        onClick = {
+                            onEvent(
+                                ProxyViewModel.Event.ProtocolTypeSelected(
+                                    protocol = text,
+                                ),
+                            )
+                        },
+                        role = Role.RadioButton,
+                    ).padding(horizontal = 16.dp),
+                    verticalAlignment = Alignment.CenterVertically,
+                ) {
+                    RadioButton(
+                        selected = (text == state.proxyType),
+                        onClick = null,
+                    )
+                    Text(
+                        text = stringResource(text.label),
+                        style = MaterialTheme.typography.bodyLarge,
+                        modifier = Modifier.padding(start = 16.dp),
+                    )
+                }
+            }
+        }
+        if (state.proxyType == ProxyType.CUSTOM) {
+            Row(
+                modifier = Modifier.fillMaxWidth().padding(8.dp),
+                horizontalArrangement = Arrangement.spacedBy(8.dp),
+            ) {
+                var expanded by remember { mutableStateOf(false) }
+                var hostname by rememberSaveable { mutableStateOf(state.proxyHost ?: "") }
+                var port by rememberSaveable { mutableStateOf(state.proxyPort?.toString() ?: "") }
+                val protocols =
+                    listOf(ProxyProtocol.HTTP, ProxyProtocol.HTTPS, ProxyProtocol.SOCKS5)
+                OutlinedTextField(
+                    value = state.proxyProtocol.toCustomProtocol(),
+                    onValueChange = { },
+                    label = { Text(stringResource(Res.string.Settings_Proxy_Custom_Protocol), maxLines = 1) },
+                    modifier = Modifier.weight(0.3f),
+                    readOnly = true,
+                    singleLine = true,
+                    maxLines = 1,
+                    trailingIcon = {
+                        IconButton(onClick = { expanded = true }) {
+                            Icon(
+                                Icons.Filled.ArrowDropDown,
+                                contentDescription = stringResource(Res.string.Settings_Proxy_Custom_Protocol),
+                            )
+                        }
+                    },
+                )
+                DropdownMenu(expanded = expanded, onDismissRequest = { expanded = false }) {
+                    protocols.forEach { protocol ->
+                        DropdownMenuItem(text = { Text(protocol.protocol) }, onClick = {
+                            expanded = false
+                            onEvent(
+                                ProxyViewModel.Event.ProtocolChanged(
+                                    protocol = protocol,
+                                ),
+                            )
+                        })
+                    }
+                }
```

Comment:
This isn't working very well. I can tap the contents of the Custom Proxy field and it goes into a focused state, but doesn't open the dropdown. Probably the whole field needs to open the dropdown on touch/focus.